src/components/carbon-provider/carbon-provider.test.js

```diff
@@ -1,11 +1,275 @@
 import React from "react";
 import * as stories from "./carbon-provider.stories";
+import Button from "../button/button.component";
+import Link from "../link/link.component";
+import Loader from "../loader/loader.component";
+import LoaderBar from "../loader-bar/loader-bar.component";
+import MultiActionButton from "../multi-action-button/multi-action-button.component";
+import Pill from "../pill/pill.component";
+import SplitButton from "../split-button/split-button.component";
+import StepSequenceItem from "../step-sequence/step-sequence-item/step-sequence-item.component";
+import { Tabs, Tab } from "../tabs";
 import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
 
+import {
+  sageTheme,
+  mintTheme,
+  noTheme,
+  aegeanTheme,
+} from "../../style/themes/index";
+import { buttonDataComponent } from "../../../cypress/locators/button";
+import {
+  linkComponent,
+  loaderBarComponent,
+  loaderComponent,
+} from "../../../cypress/locators/themes";
+import { pillPreview } from "../../../cypress/locators/pill";
+import { stepSequenceDataComponentItem } from "../../../cypress/locators/step-sequence";
+import { getDataElementByValue } from "../../../cypress/locators";
+import { multiActionButtonComponent } from "../../../cypress/locators/multi-action-button";
+import { splitMainButton } from "../../../cypress/locators/split-button";
+
+const themes = [mintTheme, aegeanTheme, noTheme, sageTheme];
+
+const buildTestArray = (array) =>
+  themes.map((theme, i) => [theme.name, theme, array[i]]);
+const buildTestDataWithTwoArrays = (firstArray, secondArray) =>
+  themes.map((theme, i) => [theme.name, theme, firstArray[i], secondArray[i]]);
+
+const commonColors = [
+  "rgb(0, 125, 90)",
+  "rgb(0, 115, 194)",
+  "rgb(0, 130, 0)",
+  "rgb(0, 126, 69)",
+];
+
+const commonColorsOnHover = [
+  "rgb(0, 96, 70)",
+  "rgb(0, 92, 154)",
+  "rgb(0, 99, 0)",
+  "rgb(0, 103, 56)",
+];
+
+const multiButtonOnHover = [
+  "rgb(0, 64, 46)",
+  "rgb(0, 68, 114)",
+  "rgb(0, 77, 42)",
+  "rgb(0, 77, 42)",
+];
+
+const stepSequnceColors = [
+  "rgb(0, 125, 90)",
+  "rgb(0, 115, 194)",
+  "rgb(0, 138, 33)",
+  "rgb(0, 138, 33)",
+];
+
+const loaderBarColors = [
+  "rgb(179, 227, 214)",
+  "rgb(179, 214, 239)",
+  "rgb(179, 224, 179)",
+  "rgb(179, 217, 200)",
+];
+
 context("Testing Carbon Provider component", () => {
-  describe.skip("should render Carbon Provider component", () => {
-    // functionality of theming is tested in `theme.test.js` file
-  });
+  describe.each(buildTestArray(commonColors))(
+    "should render components with %s theme",
+    (theme, themeName, color) => {
```

Comment:
Given the definition of `buildTestArray` on line 34, I think these parameter names need swapping around

```suggestion
    (themeName, theme, color) => {
```

Reply:
I don't think it matters :)

Reply:
it is used properly :)

Reply:
The tests still work as intended, however it is a bit misleading, as `themeName` could be read as a string when it's actually a theme object.